brave/brave6/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java

```diff
@@ -120,11 +122,13 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
         final Span span = handler.handleSend(braveReq);
         req = req.withHeaders(newHeaders);
         ctx.updateRequest(req);
+        TraceContextUtil.setTraceContext(ctx, span.context());
```

Comment:
Because a span is fixed to a `RequestContext`, how about leaving a warning log if a `RequestContext` has a `TraceContext` already? I've seen internal customers who have used this API.
Also, I think we can remove `@Nullable` from `setTraceContext` method.

Reply:
I'm unsure if leaving a log level with `WARN` level is a good idea since it is a valid behavior to do so. e.g. Using `TraceContextPropagation#inject` (span1) with `BraveClient` (span2) will also overwrite the span where `span2.parent == span1`.

What do you think of leaving a trace level log so that we can guide users if there is unexpected behavior?

Alternatively, I can also add a static field guard so that the log is left only once if you feel strongly of this.

Reply:
Ah, that's valid situtation. IIUC, in that case, the same `TraceContext` can be set to a `RequestContext`. Is it right? If so, can't we add a warning log if it tries to set a different `TraceContext`?

Reply:
> I've seen internal customers who have used this API.

It would help if you could explain why users using this API is a bad idea.  If users were setting the `TraceContext` manually before this change, couldn't they continue using it this way after this change as well?

> If so, can't we add a warning log if it tries to set a different TraceContext?

Just to be clear are you suggesting that a warning log is printed once guarded by a static variable? or every time a different context is set?

Reply:
> Just to be clear are you suggesting that a warning log is printed once guarded by a static variable? or every time a different context is set?

Modified to just leave a warning every time a different context is set, let me know if this works for you

Reply:
> It would help if you could explain why users using this API is a bad idea. If users were setting the TraceContext manually before this change, couldn't they continue using it this way after this change as well?

I've imagined some of this situation. Please let me know if this doesn't make sense:
```java
ClientRequextContext ctx = ...
TraceContextUtil.setTraceContext(ctx, span1.context());
ctx.makeContextAware(future).thenApply(
  // This handler will be executed later after `TraceContextUtil.setTraceContext(ctx, span2.context());` is invoked.
  // Expected to execute with span1 but executed with span2.
)

// Other sets different TraceContext to the ctx:
TraceContextUtil.setTraceContext(ctx, span2.context());
```

Reply:
I still maintain that 

1) the scenario you shared seems like a possible case prior to the changes in this PR
2) the warning log may print for valid scenarios mentioned in here https://github.com/line/armeria/pull/6139#discussion_r2036397465

Anyways, I've updated to always leave a warning log as you requested. PTAL when you have time

Reply:
> the warning log may print for valid scenarios mentioned in here

Since it's a valid scenario, could you change to log just one time?

Reply:
Good point, done